internal/msalbase/authentication_result.go

```diff
@@ -5,82 +5,67 @@ package msalbase
 
 import (
 	"errors"
-	"reflect"
 	"time"
 )
 
 // AuthenticationResult contains the results of one token acquisition operation in PublicClientApplication
 // or ConfidentialClientApplication. For details see https://aka.ms/msal-net-authenticationresult
 type AuthenticationResult struct {
-	Account        *Account
-	idToken        *IDToken
+	Account        Account
+	idToken        IDToken
 	AccessToken    string
 	ExpiresOn      time.Time
 	GrantedScopes  []string
 	DeclinedScopes []string
 }
 
-//CreateAuthenticationResultFromStorageTokenResponse creates an authenication result from a storage token response (which is generated from the cache)
-func CreateAuthenticationResultFromStorageTokenResponse(storageTokenResponse *StorageTokenResponse) (*AuthenticationResult, error) {
-	if storageTokenResponse == nil {
-		return nil, errors.New("storage token response is nil")
+// CreateAuthenticationResultFromStorageTokenResponse creates an authenication result from a storage token response (which is generated from the cache).
+func CreateAuthenticationResultFromStorageTokenResponse(storageTokenResponse StorageTokenResponse) (AuthenticationResult, error) {
+	if storageTokenResponse.AccessToken == nil {
+		return AuthenticationResult{}, errors.New("no access token present in cache")
 	}
+
 	account := storageTokenResponse.account
-	var idToken *IDToken
-	accessToken := ""
-	var expiresOn time.Time
-	grantedScopes := []string{}
-	declinedScopes := []string{}
-	var err error
-	// Checking if the cache had an access token, if not, need to throw an error
-	if !reflect.ValueOf(storageTokenResponse.accessToken).IsNil() {
-		accessToken = storageTokenResponse.accessToken.GetSecret()
-		expiresOn, err = ConvertStrUnixToUTCTime(storageTokenResponse.accessToken.GetExpiresOn())
-		if err != nil {
-			return nil, errors.New("access token in cache expires at an invalid time")
-		}
-		grantedScopes = SplitScopes(storageTokenResponse.accessToken.GetScopes())
-	} else {
-		return nil, errors.New("no access token present in cache")
+	accessToken := storageTokenResponse.AccessToken.GetSecret()
+	expiresOn, err := ConvertStrUnixToUTCTime(storageTokenResponse.AccessToken.GetExpiresOn())
+	if err != nil {
+		return AuthenticationResult{}, errors.New("access token in cache expires at an invalid time")
```

Comment:
>access token in cache expires at an invalid time [](start = 45, length = 48)

nit: interesting wording

Reply:
I'm not sure what you mean from the comment.  FYI errors.New() is the original error (the + probably just means I moved it around).  

Do you want:

return AuthenticationResult{}, fmt.Errorf("access token validation error on field expiresOn: %w", err)

And then go into ConvertStrUnixToUTCTime(which I am going to get rid of) and have it return:

fmt.Errorf("time field was an unrecognizable string(%s)", str)

Reply:
@henrik-me ping

Reply:
I think @henrik-me is saying that the error message is not actionable enough for a developer using MSAL.go
I do think you should necessarily try to improve error messages, because you may not have enough "MSAL context" to improve them. 

In this instance, it's possible (although highly unlikely) that a response has a "expires_in" key / pair that is invalid (missing or not a positive int). So maybe smth more in the lines of `The token response from the server is invalid because expires_in is missing / set to X`

Reply:
Error updated.